ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java

```diff
@@ -1234,4 +1236,46 @@ public void shouldParseMultiLineWithInlineBracketedComments() {
     assertThat(statements, hasSize(1));
     assertThat(statements.get(0).getStatement(), is(instanceOf(ListStreams.class)));
   }
+
+  @Test
+  public void shouldBuildSimpleCaseStatement() {
+    // Given:
+    final String statementString =
+        "CREATE STREAM S AS SELECT CASE WHEN orderunits < 10 THEN 'small' WHEN orderunits < 100 THEN 'medium' ELSE 'large' END FROM orders;";
+
+    // When:
+    final Statement statement = KSQL_PARSER.buildAst(statementString, metaStore).get(0)
+        .getStatement();
+
+    // Then:
+    final Query query = ((CreateStreamAsSelect) statement).getQuery();
+    final QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
+    final Expression caseExpression = ((SingleColumn) querySpecification.getSelect().getSelectItems().get(0)).getExpression();
+    final SearchedCaseExpression searchedCaseExpression = (SearchedCaseExpression) caseExpression;
+    assertThat(searchedCaseExpression.getWhenClauses().size(), equalTo(2));
+    assertThat(searchedCaseExpression.getWhenClauses().get(0).getOperand().toString(), equalTo("(ORDERS.ORDERUNITS < 10)"));
+    assertThat(searchedCaseExpression.getWhenClauses().get(0).getResult().toString(), equalTo("'small'"));
+    assertThat(searchedCaseExpression.getWhenClauses().get(1).getOperand().toString(), equalTo("(ORDERS.ORDERUNITS < 100)"));
+    assertThat(searchedCaseExpression.getWhenClauses().get(1).getResult().toString(), equalTo("'medium'"));
+    assertTrue(searchedCaseExpression.getDefaultValue().isPresent());
+    assertThat(searchedCaseExpression.getDefaultValue().get().toString(), equalTo("'large'"));
+  }
+
+  @Test
+  public void shouldBuildSimpleCaseWithoutDefaultStatement() {
```

Comment:
```suggestion
  public void shouldBuildSearchedCaseWithoutDefaultStatement() {
```
Rename to avoid confusion if we want to support simple case expressions in the future.